### ISSUE TITLE
🐛 lint/style/noCommaOperator incorrectly alerts on 3-update for loop

### ISSUE THREAD

**Issue by turbocrime** [biomejs/biome#138]
### What happened?

`lint/style/noCommaOperator` will alert on a for loop with three or more update expressions.

```ts
for (let i = 0, j = 1, k = 2; i < 100; i++, j++, k++) {}
```

this will alert on the comma after `i++` with default settings.

https://biomejs.dev/playground/?code=ZgBvAHIAIAAoAGwAZQB0ACAAaQAgAD0AIAAwACwAIABqACAAPQAgADEALAAgAGsAIAA9ACAAMgA7ACAAaQAgADwAIAAxADAAMAA7ACAAaQArACsALAAgAGoAKwArACwAIABrACsAKwApACAAewB9AA%3D%3D

### Expected result

`lint/style/noCommaOperator` should not apply to the top level of for loop update expressions.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
